src/main/java/com/google/devtools/build/lib/skyframe/toolchains/RegisteredExecutionPlatformsFunction.java

```diff
@@ -245,15 +246,15 @@ private static ImmutableList<TargetPattern> getBzlmodExecutionPlatforms(
       configureRegisteredExecutionPlatforms(
           Environment env, BuildConfigurationValue configuration, List<Label> labels)
           throws InterruptedException, RegisteredExecutionPlatformsFunctionException {
-    ImmutableList<ConfiguredTargetKey> keys =
+    ImmutableSet<ConfiguredTargetKey> keys =
```

Comment:
The order is important here: does ImmutableSet have a documented order? Can you leave a comment clarifying that using a set that keeps insertion order is important?

Reply:
Yes, all `Immutable*` data structures do, just like the `ImmutableMap` constructed from the set below. I'm not quite sure where to add this comment without also adding it in many other places.

Reply:
Fair enough.